app/Services/Model/Imp/SpeakerService.php

```diff
@@ -1313,116 +1308,42 @@ public function triggerSendEmails(Summit $summit, array $payload, $filter = null
-
-        $test_email_recipient = null;
-        if(isset($payload['test_email_recipient']))
-            $test_email_recipient = $payload['test_email_recipient'];
-
-        if(isset($payload['should_resend'])){
-            $speaker_announcement_email_config->setShouldResend(boolval($payload['should_resend']));
-        }
-        if(isset($payload['should_send_copy_2_submitter'])){
-            $speaker_announcement_email_config->setShouldSendCopy2Submitter(boolval($payload['should_send_copy_2_submitter']));
-        }
-
-        $page = 1;
-        $count = 0;
-        $maxPageSize = 100;
-
-        Log::debug(sprintf("SpeakerService::send summit id %s flow_event %s filter %s",
-            $summit_id, $flow_event, is_null($filter) ? '' : $filter->__toString()));
-
-        EmailExcerpt::addInfoMessage(
-            sprintf("Processing EMAIL %s for summit %s", $flow_event, $summit_id)
-        );
-
-        $summit = $this->tx_service->transaction(function () use($summit_id){
-            $summit = $this->summit_repository->getById($summit_id);
-            if (is_null($summit) || !$summit instanceof Summit) return null;
-            return $summit;
-        });
-
-        if(is_null($summit)){
-            Log::debug("SpeakerService::send summit is null");
-            return;
-        }
-
-        do {
-
-            $ids = $this->tx_service->transaction(function () use ($summit, $payload, $filter, $page, $maxPageSize) {
-                if (isset($payload['speaker_ids'])) {
-                    Log::debug(sprintf("SpeakerService::send summit id %s speakers_ids %s", $summit->getId(),
-                        json_encode($payload['speaker_ids'])));
-                    return $payload['speaker_ids'];
-                }
-                Log::debug(sprintf("SpeakerService::send summit id %s getting by filter", $summit->getId()));
-                if (is_null($filter)) {
-                    $filter = new Filter();
-                }
-
-                Log::debug(sprintf("SpeakerService::send page %s", $page));
-                return $this->speaker_repository->getSpeakersIdsBySummit($summit, new PagingInfo($page, $maxPageSize), $filter);
-            });
-
-            Log::debug(sprintf("SpeakerService::send summit id %s flow_event %s filter %s page %s got %s records",
-                $summit_id, $flow_event, is_null($filter) ? '' : $filter->__toString(), $page, count($ids)));
-
-            if (!count($ids)) {
-                // if we are processing a page, then break it
-                Log::debug(sprintf("SpeakerService::send summit id %s page is empty, ending processing.", $summit_id));
-                break;
-            }
-
-            $email_strategy = new SpeakerActionsEmailStrategy($summit, $flow_event);
-
-            foreach ($ids as $speaker_id) {
-
+        $this->_sendEmails(
+            $summit_id,
+            $this->summit_repository,
+            $payload,
+            $this->tx_service,
```

Comment:
you dont need this param, its on this context , just use it on trait as $this->tx_service